src/amp-story-player/amp-story-player-iframe-pool.js

```diff
@@ -77,31 +87,59 @@ export class IframePool {
   }
 
   /**
-   * Finds adjacent iframe indices given an index.
-   * Examples of resulting adjacent arrays:
+   * Finds adjacent iframes given an story, starting in N and going to N+1.
+   * Following a DFS order.
+   *
+   * It will only allow up to MAX_IFRAMES to be loaded. Those that can be loaded
+   * will have the `shouldLoad` property as `true`. The rest should only be
+   * positioned.
+   *
+   * Examples of traversal:
+   * idx
+   * 1 -> [2] [0] [1] [] []
    * 0 -> [0] [1] [2] [] []
-   * 1 -> [0] [1] [2] [] []
-   * 4 -> [] [] [0] [1] [2]
-   * 1 -> [0] [1]
+   * 4 -> [] [] [2] [1] [0]
+   * 0 -> [0] [1]
+   * 1 -> [1] [0]
    * @param {number} storyIdx
    * @param {number} maxIdx
-   * @return {!Array<number>}
+   * @return {!Array<!Object>}
    */
-  findAdjacent(storyIdx, maxIdx) {
+  findAdjacentDFS(storyIdx, maxIdx) {
     const adjacent = [];
+    let iframesToLoad = MAX_IFRAMES;
 
-    // If index is at rightmost part of the array, adjacent iframes will all be
-    // at the left.
-    let cursor = storyIdx + 1 > maxIdx ? storyIdx - 2 : storyIdx - 1;
-
-    // Place current story index first to prioritize loading over adjacent ones.
-    adjacent.push(storyIdx);
+    const neighborsQueue = [storyIdx, storyIdx + 1, storyIdx - 1];
     while (adjacent.length < this.iframePool_.length) {
-      if (cursor < 0 || cursor === storyIdx) {
-        ++cursor;
+      const cursorIdx = neighborsQueue.shift();
+
+      if (cursorIdx > maxIdx || cursorIdx < 0) {
+        continue;
+      }
+
+      adjacent.push({
+        storyIdx: cursorIdx,
+        shouldLoad: iframesToLoad > 0,
```

Comment:
`iframesToLoad` quota will always be 0 once you get to `storyIdx - 1`, so you only prerender when navigating forward, is it a bug or a feature?